### PR TITLE
test: checking version number is matching one in pubspec.yaml

### LINKED ISSUE #388: Add test to check version number is correct
Now that we have our tests being run for every open PRs, it would be good to have a test which ensures that both the generated `packageVersion` matches the number in the `pubspec.yaml `

If this tests fails, this would indicate that we need to run build runner again to update the `packageVersion` value

### PR DESCRIPTION
**Pull request by MarkOSullivan94**
Resolves #388

### CODE REVIEW

File: lib/pubspec_parser.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/// helper class for parsing the contents of pubspec file
+class PubspecParser {
+
+  /// ensures unnamed constructor cannot be used as this class should only have
+  /// static methods
+  PubspecParser._();
+
+  /// parses the pubspec located at [path] to map
+  static Map toMap(String path) {
```
Comment (RatakondalaArun) [inline lib/pubspec_parser.dart:13]: I suggest to rename `toMap()` function to `fromFSPath` or `fromPath` or `loadFromPath`

File: lib/main.dart [hunk 2]
```diff
@@ -197,10 +198,7 @@ Map<String, dynamic>? loadConfigFileFromArgResults(ArgResults argResults, {bool
 }
 
 Map<String, dynamic> loadConfigFile(String path, String? fileOptionResult) {
-  final File file = File(path);
-  final String yamlString = file.readAsStringSync();
-  // ignore: always_specify_types
-  final Map yamlMap = loadYaml(yamlString);
+  final yamlMap = PubspecParser.toMap(path);
 
   if (!(yamlMap['flutter_icons'] is Map)) {
```
Comment (RatakondalaArun) [inline lib/main.dart:203]:
```suggestion
  if (yamlMap['flutter_icons'] is! Map) {
```

[`prefer_is_not_operator`](https://dart.dev/tools/linter-rules#prefer_is_not_operator)

**Review by RatakondalaArun — CHANGES_REQUESTED** [review]

File: lib/pubspec_parser.dart (hunk 1 above)
Comment (MarkOSullivan94) [reply lib/pubspec_parser.dart:13]: What about `fromPathToMap(...)`?

File: lib/main.dart (hunk 2 above)
Comment (MarkOSullivan94) [reply lib/main.dart]: Updated

File: lib/pubspec_parser.dart (hunk 1 above)
Comment (RatakondalaArun) [reply lib/pubspec_parser.dart:13]: Even better

Comment (MarkOSullivan94) [reply lib/pubspec_parser.dart]: Done!

**Review by RatakondalaArun — APPROVED** [review]
LGTM 🚀

**Pull request merged**